core/impl/src/main/java/com/blazebit/expression/impl/PredicateModelGenerator.java

```diff
@@ -488,6 +490,18 @@ public Expression visitCollectionLiteral(PredicateParser.CollectionLiteralContex
 
     @Override
     public Expression visitPath(PredicateParser.PathContext ctx) {
+        return createPathExpression(ctx);
+    }
+    @Override
+    public Expression visitPathPredicate(PathPredicateContext ctx) {
+        Expression expression = createPathExpression(ctx.path());
+        DomainType type = expression.getType();
+        if (!type.equals(getBooleanDomainType())) {
```

Comment:
Please also add a `visitPathPredicate` method in `editor/monaco/src/main/typescript/index.ts#MyBlazeExpressionParserVisitor` that asserts the domain type similar to what `visitBooleanFunction` does.